pkg/collector/tcp.go

```diff
@@ -79,31 +82,39 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 				return
 			}
 			klog.V(2).Infof("Receiving %d bytes from %s", size, address)
-			buffBytes := make([]byte, size)
-			copy(buffBytes, buff[:size])
+			klog.V(2).Infof("Remaining bytes from last read length: %v", len(remainingBytes))
+			size = size + len(remainingBytes)
+			copiedBytes := make([]byte, size)
+			copy(copiedBytes[:len(remainingBytes)], remainingBytes)
+			copy(copiedBytes[len(remainingBytes):], buff[:size-len(remainingBytes)])
```

Comment:
this whole function is too complicated in my mind and with too many data copies. I think you should use a `bufio.Reader` object for performance and to abstract away the socket.

```go
buff := make([]byte, cp.maxBufferSize)
r := bufio.NewReader(conn)
for {
    length, version, err := getMessageLength(r)
    // ...

    // read the full message
    _, err := io.ReadFull(r, buff[:length])
    // ...

    // at this stage you have one IPFIX message stored at the beginning of `buff`
}
```

Reply:
Thanks @antoninbas for the suggestion. It simplified the implementation logic quite a bit.
However, I feel we are reading bytes of the packet twice. First into the buffer from `bufio.Reader` and then next from the `buffer.Bytes` to decode the packet. I feel there is room for improvement there. Any thoughts?